### PR TITLE
Feature/redux favorite advanced edit

### LINKED ISSUE #61: As a user, I want to update a puppy
EPIC: Build `redux_favorite_advanced_sample` with the use of [Redux](https://flutter.dev/docs/development/data-and-backend/state-mgmt/options#redux) as a state management solution.
DevNote: [favorites_advanced_base](https://github.com/Prime-Holding/rx_bloc/tree/master/examples/favorites_advanced/favorites_advanced_base) should be used to reduce code repetition.

- [x] **As a user, I want to update a puppy**
  - [x] Once I tap on the pencil icon I should be navigated to the Puppy Edit Page
  - [x] Validations && pre-populating fields  as described [here](https://medium.com/prime-holding-jsc/building-forms-in-flutter-454b8d65f48)
  - [x] I should be able to update the following properties 
    - [x] pic: Camera && Gallery
    - [x] Name: Text field
    - [x] Breed: Select 
    - [x] Gender: Radio buttons
    - [x] Characteristics: Multiline textfield
  - [x] By default the save button should be enabled, only once any of the fields have changed the button should become enabled.
  - [x] Once I save the puppy, by tapping on the save button, a loading indicator should appear (as per design) and I should be navigated back to the details screen. 
    - [x] A snackbar should appear so I know the puppy have been successfully saved.
    - [x] All lists containing the puppy should be updated accordingly.
    - [x] In case of no internet connection a snackbar should appear just after the user has pressed the save button. He/she should stay on the page as all the user input/pre-filled data should be preserved.
- [x] Unit tests

Estimated date: 21.5.2021.

**GIT: Branching strategy:**
1. `feature/redux_favorite_advanced` will be the `develop` branch for this epic
2. For each story a dedicated branch needs to be created (for instance `feature/redux_favorite_advanced/update`). Once a story is being completed a PR should be opened against the main branch (`feature/redux_favorite_advanced`)

### PR DESCRIPTION
**Pull request by GoranPrime**
Closes #61

### CODE REVIEW

File: examples/favorites_advanced/redux_favorites_advanced/lib/feature_puppy/edit/ui_components/puppy_edit_form.dart [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                    key: const ValueKey('PuppyNameInputField'),
+                    maxLines: 1,
+                    initialValue: _viewModel.puppy.name,
+                    cursorColor: const Color(0xff333333),
+                    style: TextStyles.editableTextStyle,
+                    textInputAction: TextInputAction.next,
+                    maxLengthEnforcement: MaxLengthEnforcement.enforced,
+                    decoration: InputStyles.textFieldDecoration,
+                    onChanged: (name) => _viewModel.onNameChange(name),
+                    validator: (value) => (value == null || value.isEmpty)
```
Comment (StanevPrime) [inline examples/favorites_advanced/redux_favorites_advanced/lib/feature_puppy/edit/ui_components/puppy_edit_form.dart:53]: @GoranPrime, please move all validation logic to the business layer

File: examples/favorites_advanced/redux_favorites_advanced/lib/feature_puppy/edit/ui_components/puppy_edit_form.dart [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                    maxLines: 8,
+                    initialValue: _viewModel.puppy.breedCharacteristics,
+                    cursorColor: const Color(0xff333333),
+                    style: TextStyles.editableTextStyle,
+                    textInputAction: TextInputAction.done,
+                    maxLengthEnforcement: MaxLengthEnforcement.enforced,
+                    decoration: InputStyles.textFieldDecoration,
+                    onChanged: (characteristics) =>
+                        _viewModel.onCharacteristicsChange(characteristics),
+                    validator: (value) => (value == null || value.isEmpty)
```
Comment (StanevPrime) [inline examples/favorites_advanced/redux_favorites_advanced/lib/feature_puppy/edit/ui_components/puppy_edit_form.dart:145]: @GoranPrime, please move all validation logic to the business layer

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: examples/favorites_advanced/redux_favorites_advanced/lib/feature_puppy/edit/ui_components/puppy_edit_form.dart (hunk 1 above)
Comment (GoranPrime) [reply examples/favorites_advanced/redux_favorites_advanced/lib/feature_puppy/edit/ui_components/puppy_edit_form.dart]: @StanevPrime Corrected

File: examples/favorites_advanced/redux_favorites_advanced/lib/feature_puppy/edit/ui_components/puppy_edit_form.dart (hunk 2 above)
Comment (GoranPrime) [reply examples/favorites_advanced/redux_favorites_advanced/lib/feature_puppy/edit/ui_components/puppy_edit_form.dart]: @StanevPrime Corrected

**Comment by StanevPrime** [comment]
@GoranPrime, could you please resolve the conflicts, so I can merge it ?

**Comment by GoranPrime** [comment]
> @GoranPrime, could you please resolve the conflicts, so I can merge it ?

@StanevPrime done

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**